src/server/svr_jobfunc.c

```diff
@@ -3029,11 +3029,31 @@ Time4occurrenceFinish(resc_resv *presv)
 	/* If the start time of the reservation was altered, copy from RESV_ATR_start
  	 * will make the next instance to have it's start time altered so take the start
  	 * time from the ri_alter_stime. */
-	if (presv->ri_alter_stime) {
-		dtstart = presv->ri_alter_stime;
-		presv->ri_alter_stime = 0;
+	if (presv->ri_alter.ra_revert.rr_stime) {
```

Comment:
I think it is better to check for flags than the value. What do you think?

Reply:
There are two parts of the ra_alter structure.  There are elements to keep track of the current values, and a structure to keep track of the original values.  The current values are used to revert a reservation back when the alter fails.  The ra_revert is used for reverting subsequent occurrences of a standing reservation.  When we alter a standing reservation, we only alter the current occurrence.  We need to keep track of the original values because we might alter the same occurrence multiple times.